src/raytracer/Raytracer.m

```diff
@@ -165,126 +157,290 @@
+            'Rotation', [nodeCfgInput.nodeEquivalentRotationAngle(:,:,i); [inf inf inf]]);
+        json = jsonencode(s); % Add a temporary inf vector to make sure
+        % more than a single vector will be encoded. Matlab json
+        % encoder loses the square brackets when encoding vectors.
+        str2remove =',[null,null,null]'; %Temporary string to remove
+        rem_ind_start = num2cell(strfind(json, str2remove)); % Find start string to remove
+        index2rm = cell2mat(cellfun(@(x) x:x+length(str2remove)-1,rem_ind_start,'UniformOutput',false)); % Create index of char to remove
+        json(index2rm) = []; % Remove temporary vector.
+        fprintf(f, '%s\n', json);
+    end
+    fclose(f);
+    
+    % Write PAAPosition.json
+    f = fopen(strcat(visualizerPath, filesep,'PAAPosition.json'), 'w');
+    for i = 1:paraCfgInput.numberOfNodes
+        idOrientation = 0;
+        for paaId = 1:nPAA_centroids(i)
+            nodeTxCluster  = nodeCfgInput.PAA_info{i}.paaInCluster{paaId};
+            for paaCentroid = 1:length(nodeTxCluster)
+                idOrientation = idOrientation+1;
+                s = struct('Node', i-1, 'PAA',nodeTxCluster(paaCentroid)-1, ...
+                    'Centroid', nodeCfgInput.PAA_info{i}.centroids(paaId)-1, ...
+                    'Orientation', nodeCfgInput.nodeAntennaOrientation{i}(idOrientation,:) ,...
+                'Position', [reshape(squeeze(nodeCfgInput.PAA_info{i}.centroid_position_rot(:,paaId,:)), [],3); [inf inf inf]]);
+                json = jsonencode(s);% Add a temporary inf vector to make sure
+                % more than a single vector will be encoded. Matlab json
+                % encoder loses the square brackets when encoding vectors.
+                str2remove =',[null,null,null]'; %Temporary string to remove
+                rem_ind_start = num2cell(strfind(json, str2remove)); % Find start string to remove
+                index2rm = cell2mat(cellfun(@(x) x:x+length(str2remove)-1,rem_ind_start,'UniformOutput',false)); % Create index of char to remove
+                json(index2rm) = []; % Remove temporary vector.
+                fprintf(f, '%s\n', json);
+            end
+        end
+    end
+    fclose(f); 
+end
+
+if ~paraCfgInput.jsonOutput || keepBothQDOutput
+    closeQdFilesIds(fids, paraCfgInput.useOptimizedOutputToFile);
+end
 
+%% Write useful output information. 
+writeReportOutput =0 ; %Set to 0 to allow succeful test.
+if writeReportOutput
+    f = fopen(strcat(outputPath, filesep,'report.dat'), 'w'); %#ok<UNRCH>
+%     elapsedTime = toc;
+%     fprintf(f, 'Elapsed Time:\t%f\n', elapsedTime);    
+    fprintf(f, 'Device Rotation:\t%d\n', paraCfgInput.isDeviceRotationOn);
+%     isInitialOrientationOn = 'true';
+    fprintf(f, 'Initial Orientation:\t%d\n', paraCfgInput.isInitialOrientationOn);
+%     isPAAcentered = 'true';
+    fprintf(f, 'PAA centered:\t%d\n', paraCfgInput.isPAAcentered);
+    fclose(f);
```

Comment:
This part is disabled to allow the previous test to be successful (number of file count). It creates a file with useful information. I think it's useful and it can be filled with stats in future to have a quick overview of the channel that has been generated.